test/firebase_auth_mocks_test.dart

```diff
@@ -344,6 +345,61 @@ void main() {
     final authExceptions2 = AuthExceptions();
     expect(authExceptions1, authExceptions2);
   });
+
+  test('Id token contains user data', () async {
+    final idToken = await tUser.getIdToken();
+    final decodedToken = JwtDecoder.decode(idToken);
+    expect(decodedToken['name'], tUser.displayName);
+    expect(decodedToken['picture'], tUser.photoURL);
+    expect(decodedToken['user_id'], tUser.uid);
+    expect(decodedToken['sub'], tUser.uid);
+    expect(decodedToken['email'], tUser.email);
+    expect(decodedToken['email_verified'], tUser.emailVerified);
+  });
+
+  test('Id token contains user data when using default value', () async {
+    final user = MockUser();
+    final idToken = await user.getIdToken();
+    final decodedToken = JwtDecoder.decode(idToken);
+    expect(decodedToken['name'], user.displayName);
+    expect(decodedToken['picture'], user.photoURL);
+    expect(decodedToken['user_id'], user.uid);
+    expect(decodedToken['sub'], user.uid);
+    expect(decodedToken['email'], user.email);
+    expect(decodedToken['email_verified'], user.emailVerified);
+  });
+
+  test('getIdToken still works when using the default value', () async {
+    final idToken = await MockUser().getIdToken();
+    final decodedToken = JwtDecoder.decode(idToken);
+    expect(decodedToken['name'] != null, true);
```

Comment:
Prefer `isNotNull` instead of comparing this to null. This gives slightly more readable error messages if the unit test were to fail.

Reply:
Ok I got it